package/lib/src/beam_update_guard.dart

```diff
@@ -0,0 +1,72 @@
+import 'dart:async';
+
+import 'package:beamer/beamer.dart';
+import 'package:beamer/src/beam_guard_util.dart';
+import 'package:flutter/widgets.dart';
+
+/// Checks whether current [RouteInformation] is allowed to be beamed to
+/// and provides steps to be executed following a failed check.
+class BeamUpdateGuard {
```

Comment:
I would call it just `UpdateGuard`. Then we can also rename current `BeamGuard` to `BuildGuard`.
(the end goal is too have async `UpdateGuard`s and non-async `BuildGuard`s, which should then be obvious from "build" in the name, as build is never async)